src/Microsoft.AspNet.OData.Shared/Query/Expressions/AggregationBinder.cs

```diff
@@ -233,37 +237,131 @@ private IQueryable BindSelect(IQueryable grouping)
             return wrapperTypeMemberAssignments;
         }
 
-        private Expression CreateAggregationExpression(ParameterExpression accum, AggregateExpression expression)
+        private Expression CreateAggregationExpression(ParameterExpression accum, AggregateExpressionBase expression, Type baseType)
+        {
+            switch (expression.AggregateKind)
+            {
+                case AggregateExpressionKind.PropertyAggregate:
+                    return CreatePropertyAggregateExpression(accum, expression as AggregateExpression, baseType);
+                case AggregateExpressionKind.EntitySetAggregate:
+                    return CreateEntitySetAggregateExpression(accum, expression as EntitySetAggregateExpression, baseType);
+                default:
+                    throw new ODataException("This type of aggregation is not supported.");
```

Comment:
We have a section for exception messages:
src/Microsoft.AspNet.OData.Shared/Common/SRResources.Designer.cs
src/Microsoft.AspNet.OData.Shared/Common/SRResources.resx

Could we also include the AggregateKind in the exception message?

Reply:
also include the type information in the error message.

---
In reply to: [198944947](https://github.com/OData/webapi/pull/906#discussion_r198944947) [](ancestors = 198944947)